pkg/docker/docker.go

```diff
@@ -352,39 +355,43 @@ func (d *stiDocker) RunContainer(opts RunContainerOptions) (err error) {
 		return err
 	}
 
-	// base directory for all STI commands
-	var commandBaseDir string
-	// untar operation destination directory
-	tarDestination := opts.Destination
-	if len(tarDestination) == 0 {
-		tarDestination = getDestination(imageMetadata)
+	config := docker.Config{
+		Image: image,
 	}
-	if opts.ExternalScripts {
-		// for external scripts we must always append 'scripts' because this is
-		// the default subdirectory inside tar for them
-		commandBaseDir = filepath.Join(tarDestination, "scripts")
-		glog.V(2).Infof("Both scripts and untarred source will be placed in '%s'", tarDestination)
-	} else {
-		// for internal scripts we can have separate path for scripts and untar operation destination
-		scriptsURL := opts.ScriptsURL
-		if len(scriptsURL) == 0 {
-			scriptsURL = getScriptsURL(imageMetadata)
+
+	tarDestination := ""
+	if len(opts.Command) != 0 {
```

Comment:
i wouldn't overload the meaning of an empty opts.Command to mean "running the target image" because we might in the future want to support passing a command to run in the target image.  ie --run --command=<somecommand> instead of only running the default image CMD/ENTRYPOINT.